edweiss-app/__test__/showtime/LectureScreen.test.tsx

```diff
@@ -1,180 +1,277 @@
+}));
 
-// Mock react-native-blob-util to prevent NativeEventEmitter errors
-jest.mock('react-native-blob-util', () => ({
-    fs: {
-        dirs: {
-            DocumentDir: 'mockDocumentDir',
-        },
-    },
-    config: jest.fn(() => ({
-        fetch: jest.fn(),
-    })),
+jest.mock('react-native/Libraries/Settings/Settings', () => ({
+    get: jest.fn(),
+    set: jest.fn(),
 }));
 
-// Mock Firebase Messaging to avoid NativeEventEmitter errors
-jest.mock('@react-native-firebase/messaging', () => ({
-    __esModule: true,
-    default: jest.fn(() => ({
-        onMessage: jest.fn(),
-        onNotificationOpenedApp: jest.fn(),
-        getInitialNotification: jest.fn(),
-    })),
+jest.mock('@react-native-async-storage/async-storage', () => ({
+    setItem: jest.fn(),
+    getItem: jest.fn(),
+    removeItem: jest.fn(),
 }));
 
-// Mock Data
-const mockLectureData = {
-    data: {
-        pdfUri: 'mock-pdf-uri',
-        audioTranscript: ['Transcript for Page 1', 'Transcript for Page 2'],
-        questions: [
-            { id: '1', question: 'Question 1' },
-            { id: '2', question: 'Question 2' },
-        ],
+// Application Route
+jest.mock('@/constants/Component', () => ({
+    ApplicationRoute: jest.fn(),
+}));
+
+jest.mock('@/utils/time', () => ({
+    Time: {
+        fromDate: jest.fn(),
     },
-};
+}));
+
+jest.mock('@/components/core/containers/TView.tsx', () => {
+    const { View } = require('react-native');
+    return (props: ViewProps) => <View {...props} />;
+});
+jest.mock('@/components/core/TText.tsx', () => {
+    const { Text } = require('react-native');
+    return (props: TextProps) => <Text {...props} />;
+});
+jest.mock('@/components/core/containers/TTouchableOpacity.tsx', () => {
+    const { TouchableOpacity, View } = require('react-native');
+    return (props: React.PropsWithChildren<TouchableOpacityProps>) => (
+        <TouchableOpacity {...props}>
+            <View>{props.children}</View>
+        </TouchableOpacity>
+    );
+});
+
+jest.doMock('react-native-gesture-handler', () => {
+    return {
+        Swipeable: ({ onSwipeableOpen, children }: { onSwipeableOpen: Function, children: React.ReactNode; }) => {
+            return (
+                <TView
+                    data-testid="swipe-component"
+                    onTouchStart={() => {
+                        if (onSwipeableOpen) {
+                            onSwipeableOpen('right'); // Simuler un swipe à droite
+                        }
+                    }}
+                >
+                    {children}
+                </TView>
+            );
+        }
+    };
+});
+
+// Mock for @expo/vector-icons to render a div with accessibility label
+jest.mock('@expo/vector-icons', () => {
+    const React = require('react');
+    return {
+        Ionicons: ({ name, onPress, ...props }: { name: string, onPress?: () => void }) => (
+            <div {...props} onClick={onPress} data-testid={name} aria-label={name} />
```

Comment:
Why did you use a div element? This seems strange, as we are in React-native.

Reply:
Yes you are right, this could have been a TView instead to maintain consistency thanks